### PR TITLE
Add to Cart broken on lists view with redirect to cart enabled.

### LINKED ISSUE #32232: Add to Cart broken on lists view with redirect to cart enabled
### Preconditions (*)

1. Magento 2.4.2
1. Redirect to cart enabled

### Steps to reproduce (*)

1. Fresh Magento 2.4.2 with sample data
1. Enable redirect to cart: https://docs.magento.com/user-guide/sales/cart-redirect.html
1. Go to a list view in any of the categories and try to add an item to cart from that page

### Expected result (*)

1. Add to cart button is enabled after page finishing loading

### Actual result (*)

1. Add to cart button continues to be disabled

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [ ] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [x] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32232

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/32232

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
LGTM from first loook. Let's wait for test results

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest">
+        <annotations>
+            <stories value="Shopping Cart"/>
```
Comment (zakdma) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:11]: Add features annotation with value *Checkout*

File: app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontAddSimpleProductToCartFromComparisonListActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            <description>Add simple product from comparison page to the cart. Starts on products comparison page.</description>
+        </annotations>
+        <arguments>
+            <argument name="productName" type="string" defaultValue="{{SimpleProduct.name}}"/>
+        </arguments>
+
+        <waitForElement selector="{{StorefrontProductCompareMainSection.ProductAddToCartButton(productName)}}" stepKey="waitForAddToCartButton"/>
+        <click selector="{{StorefrontProductCompareMainSection.ProductAddToCartButton(productName)}}" stepKey="clickAddToCartButton"/>
+        <waitForElementVisible selector="{{StorefrontCategoryMainSection.SuccessMsg}}" stepKey="waitForSuccessMessage"/>
+        <see selector="{{StorefrontCategoryMainSection.SuccessMsg}}" userInput="You added {{productName}} to your shopping cart." stepKey="assertSuccessMessage"/>
```
Comment (zakdma) [inline app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontAddSimpleProductToCartFromComparisonListActionGroup.xml]: This and prev line
Use StorefrontMessagesSection.success instead

File: app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductCompareMainSection.xml [hunk 3]
```diff
@@ -14,5 +14,6 @@
         <element name="ProductPriceByName" type="text" selector="//*[@id='product-comparison']//td[.//strong[@class='product-item-name']/a[contains(text(), '{{var1}}')]]//span[@class='price']" parameterized="true"/>
         <element name="ProductImageByName" type="text" selector="//*[@id='product-comparison']//td[.//strong[@class='product-item-name']/a[contains(text(), '{{var1}}')]]//img[@class='product-image-photo']" parameterized="true"/>
         <element name="ProductAttributeByCodeAndProductName" type="text" selector="//*[@id='product-comparison']//tr[.//th[./span[contains(text(), '{{var1}}')]]]//td[count(//*[@id='product-comparison']//tr//td[.//strong[@class='product-item-name']/a[contains(text(), '{{var2}}')]]/preceding-sibling::td)+1]/div" parameterized="true"/>
+        <element name="ProductAddToCartButton" type="button" selector=".product-item-photo[title='{{productName}}']  ~ .product-item-actions button[type='submit']" parameterized="true"/>
```
Comment (zakdma) [inline app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductCompareMainSection.xml]: Please add timeout to button element

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            <title value="Add simple product to shopping cart with 'redirect to shopping cart' enabled."/>
+            <description value="Verify, user able add simple product to shopping cart from category page and compare products page when 'redirect to shopping cart' is enabled."/>
+            <testCaseId value="MC-41079"/>
+            <useCaseId value="MC-40983"/>
+            <severity value="CRITICAL"/>
+            <group value="shoppingCart"/>
```
Comment (zakdma) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:18]: add group *checkout*

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            <magentoCLI command="config:set {{DisableRedirectToShoppingCart.path}} {{DisableRedirectToShoppingCart.value}}" stepKey="disableRedirectToShippingCart"/>
+            <!--Delete test data.-->
+            <deleteData createDataKey="product" stepKey="deleteSimpleProduct"/>
+            <deleteData createDataKey="category" stepKey="deleteCategory"/>
+            <magentoCron groups="index" stepKey="reindexInvalidatedIndices"/>
+        </after>
+
+        <!--Try to add simple product to shopping cart.-->
+        <actionGroup ref="StorefrontNavigateToCategoryUrlActionGroup" stepKey="goToCategoryPage">
+            <argument name="categoryUrl" value="$category.name$"/>
```
Comment (zakdma) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml]: Please use category url_key not name

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            <argument name="productUrl" value="$product.custom_attributes[url_key]$"/>
+        </actionGroup>
+        <actionGroup ref="StorefrontAddProductToCompareActionGroup" stepKey="addProductToCompare">
+            <argument name="productVar" value="$product$"/>
+        </actionGroup>
+        <!--Try to add simple product to shopping cart from compare products page.-->
+        <actionGroup ref="SeeProductInComparisonListActionGroup" stepKey="checkProductInComparisonList">
+            <argument name="productVar" value="$product$"/>
+        </actionGroup>
+        <actionGroup ref="StorefrontAddSimpleProductToCartFromComparisonListActionGroup" stepKey="addProductToCartFromComparisonList">
```
Comment (zakdma) [inline app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:59]: Please check that redirected to cart page according to step 6 of zephyr

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:11]: Done

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml (hunk 4 above)
Comment (engcom-Kilo) [reply app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:18]: Done

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml (hunk 6 above)
Comment (engcom-Kilo) [reply app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml:59]: Done

File: app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontAddSimpleProductToCartFromComparisonListActionGroup.xml (hunk 2 above)
Comment (engcom-Kilo) [reply app/code/Magento/Checkout/Test/Mftf/ActionGroup/StorefrontAddSimpleProductToCartFromComparisonListActionGroup.xml]: Done

File: app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml (hunk 5 above)
Comment (engcom-Kilo) [reply app/code/Magento/Checkout/Test/Mftf/Test/StorefrontAddSimpleProductToCartWithRedirectToShoppingCartTest.xml]: Done

File: app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductCompareMainSection.xml (hunk 3 above)
Comment (engcom-Kilo) [reply app/code/Magento/Catalog/Test/Mftf/Section/StorefrontProductCompareMainSection.xml]: Done

**Review by zakdma — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8828](https://jira.corp.magento.com/browse/ENGCOM-8828) has been created to process this Pull Request

**Comment by dmitriyprime** [comment]
:heavy_check_mark:  QA passed

**Before:**
before

**After:**
after

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento run all tests

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32301`.

**Comment by ihor-sviziev** [comment]
@magento run Database Compare

**Pull request merged**